packages/flutter_driver/test/flutter_driver_test.dart

```diff
@@ -1213,31 +804,191 @@ String _checkAndEncode(dynamic script) {
+    if (failOnResumeWith101) {
+      throw vms.RPCError('resume', 101, '');
+    }
+    return vms.Success();
+  }
+
+  @override
+  Future<vms.Success> streamListen(String streamId) async {
+    connectionLog.add('streamListen $streamId');
+    return vms.Success();
+  }
+
+  @override
+  Future<vms.Success> streamCancel(String streamId) async {
+    connectionLog.add('streamCancel $streamId');
+    return vms.Success();
+  }
+
+  @override
+  Future<vms.Response> setFlag(String name, String value) async {
+    connectionLog.add('setFlag $name $value');
+    if (failOnSetFlag) {
+      throw Exception('setFlag failed');
+    }
+    return vms.Success();
+  }
+
+  @override
+  Stream<vms.Event> get onIsolateEvent async* {
+    connectionLog.add('onIsolateEvent');
+    yield vms.Event(
+      kind: vms.EventKind.kServiceExtensionAdded,
+      extensionRPC: 'ext.flutter.driver',
+      timestamp: 0,
+    );
+  }
+
+  List<String> commandLog = <String>[];
+  Map<String, vms.Response> responses = <String, vms.Response>{};
+  Future<void> artificialExtensionDelay;
+
+  @override
+  Future<vms.Response> callServiceExtension(String method, {Map<dynamic, dynamic> args, String isolateId}) async {
+    commandLog.add('$method $args');
+    await artificialExtensionDelay;
+
+    final vms.Response response = responses[args['command']];
+    assert(response != null, 'Failed to create a response for ${args['command']}');
+    return response;
+  }
+
+  @override
+  Future<vms.Success> clearVMTimeline() async {
+    connectionLog.add('clearVMTimeline');
+    return vms.Success();
+  }
+
+  @override
+  Future<vms.FlagList> getFlagList() async {
+    connectionLog.add('getFlagList');
+    return vms.FlagList(flags: <vms.Flag>[]);
+  }
+
+  int vmTimelineMicros = -1000000;
+  bool incrementMicros = false;
+
+  @override
+  Future<vms.Timestamp> getVMTimelineMicros() async {
+    connectionLog.add('getVMTimelineMicros');
+    if (incrementMicros || vmTimelineMicros < 0) {
+      vmTimelineMicros = vmTimelineMicros + 1000001;
+    }
+    return vms.Timestamp(timestamp: vmTimelineMicros);
+  }
+
+  @override
+  Future<vms.Success> setVMTimelineFlags(List<String> recordedStreams) async {
+    connectionLog.add('setVMTimelineFlags $recordedStreams');
+    return vms.Success();
+  }
+
+  final Map<int, vms.Timeline> timelineResponses = <int, vms.Timeline>{
+    1: vms.Timeline.parse(<String, dynamic>{
+      'traceEvents': <dynamic>[
+        <String, dynamic>{
+          'name': 'test event',
+        },
+      ],
+      'timeOriginMicros': 0,
+      'timeExtentMicros': 999999,
+    }),
+  };
+
+  @override
+  Future<vms.Timeline> getVMTimeline({int timeOriginMicros, int timeExtentMicros}) async {
+    connectionLog.add('getVMTimeline $timeOriginMicros $timeExtentMicros');
+    final vms.Timeline timeline = timelineResponses[timeOriginMicros ?? 1];
+    assert(timeline != null, 'Missing entry in timelineResponses[$timeOriginMicros]');
```

Comment:
Do we run with assertions enabled in tests? If we're not sure, maybe it's safer to just throw here.

Reply:
Assertions are enabled for tests, yes.